### PR TITLE
Change ensembler_timeout, create constants for cluster_method and add test for construct_references

### PR DESCRIPTION
**Pull request by linucks**

### CODE REVIEW

File: ample/ensembler/abinitio.py [hunk 1]
```diff
@@ -38,40 +39,40 @@ def __init__(self, **kwargs):
...
         """
-        
+
         # Cluster our protein structures
         logger.info('Generating %d clusters using method: %s', num_clusters, cluster_method)
 
         if cluster_method != 'import' and not len(models):
-            raise RuntimeError, "Cannot find any models for ensembling!" 
-        
+            raise RuntimeError, "Cannot find any models for ensembling!"
```
Comment (fsimkovic) [inline ample/ensembler/abinitio.py]: Make Python3 compatible
```python
raise RuntimeError("Cannot find any models for ensembling!")
```

File: ample/util/argparse_util.py [hunk 2]
```diff
@@ -217,3 +217,23 @@ def add_ensembler_options(parser=None):
     ensembler_group.add_argument('-truncation_scorefile', help="CSV file containing per residue scores - COLUMN ONE MUST BE RESIDUE INDEX STARTING FROM 1")
     ensembler_group.add_argument('-truncation_scorefile_header', nargs='+', help="column headers to be used to create ensembles")
     return parser
+
+def process_command_line(args=None, contacts=True, modelling=True, mol_rep=True):
+    """Process the command-line for the main AMPLE program.
+    :args: optional argument that can hold the command-line arguments if we
+    have been called from within python for testing
+    """
+    parser = argparse.ArgumentParser(
```
Comment (fsimkovic) [inline ample/util/argparse_util.py:226]: We should include the [`formatter_class=argparse.ArgumentDefaultsHelpFormatter`](https://docs.python.org/2/library/argparse.html#argparse.ArgumentDefaultsHelpFormatter) keyword, so `--help` calls show defined default values without us having to define them in the docstring of the argument.

File: ample/ensembler/__init__.py [hunk 3]
```diff
@@ -20,6 +20,7 @@
 import sys
 
 from abinitio import AbinitioEnsembler
+from constants import SPICKER_TM
```
Comment (fsimkovic) [inline ample/ensembler/__init__.py]: Maybe specify explicitly which constants we're important, i.e. `from ample.ensembler.constants ...`

**Review by fsimkovic — CHANGES_REQUESTED** [review]

File: ample/util/argparse_util.py (hunk 2 above)
Comment (linucks) [reply ample/util/argparse_util.py:226]: I'm not sure about this. We don't set defaults using argparse so this just prints lots of None's and so isn't adding anything helpful.

**Comment by linucks** [comment]
Thanks for the helpful review @fsimkovic. I've pushed some additional changes

File: ample/util/argparse_util.py (hunk 2 above)
Comment (fsimkovic) [reply ample/util/argparse_util.py:226]: Should encourage us to change the defaults to the argparser ... but fair enough, didn't know many we set

**Review by fsimkovic — APPROVED** [review]

**Pull request merged**